tests/frontend/test_volume.py

```diff
@@ -0,0 +1,148 @@
+@pytest.mark.dispense
+class TestDispense(FrontEndBase):
+
+    def test_basic(self, get_visitor):
+        file = "test_cases/volume/dispense_volume.bs"
+
+        tree = get_visitor(file)
+
+        vol = self.get_volume(tree, file)
+
+        assert vol[0] == False
+        assert sum(vol[1]['a1']['volumes']) == 50
+
+
+
+
+@pytest.mark.frontend
+@pytest.mark.volume
+@pytest.mark.mix
+class TestMix(FrontEndBase):
+
+    def test_basic(self, get_visitor):
+        file = "test_cases/volume/mix_basic.bs"
+
+        tree = get_visitor(file)
+
+        vol = self.get_volume(tree, file)
+
+        assert vol[0] == False
+
+        assert sum(vol[1]['a1']['volumes']) == -1
+        assert sum(vol[1]['b1']['volumes']) == -1
+        assert sum(vol[1]['c1']['volumes']) == 20
+
+    def test_offset(self, get_visitor):
+        file = "test_cases/volume/mix_single_offset_use.bs"
+        tree = get_visitor(file)
+
+        vol = self.get_volume(tree, file)
+
+        assert vol[0] == False
+        assert sum(vol[1]['a1']['volumes']) == -1
+        assert sum(vol[1]['a_21']['volumes']) == -1
+        assert sum(vol[1]['a_31']['volumes']) == -1
+        assert vol[1]['a_s1']['volumes'][0] == -1
+        assert vol[1]['a_s1']['volumes'][1] == 10
+        assert sum(vol[1]['b1']['volumes']) == -1
+        assert sum(vol[1]['c1']['volumes']) == 20
+
+    def test_double_var_use(self, get_visitor):
+        file = "test_cases/volume/mix_var_double_use.bs"
+        tree = get_visitor(file)
+
+        vol = self.get_volume(tree, file)
+
+        assert vol[0] == True
+
+
+class TestSplit(FrontEndBase):
+
+    def test_basic(self, get_visitor):
+        file = "test_cases/volume/split_basic.bs"
+
+        tree = get_visitor(file)
+
+        vol = self.get_volume(tree, file)
+
+        assert vol[0] == False
+        assert sum(vol[1]['a1']['volumes']) == -1
+        assert sum(vol[1]['b1']['volumes']) == 10
+
+        for i in range(5):
+            assert vol[1]['b1']['volumes'][i] == 2
+
+    def test_double_use(self, get_visitor):
+        file = "test_cases/volume/split_double_use.bs"
+
+        tree = get_visitor(file)
+
+        vol = self.get_volume(tree, file)
+
+        assert vol[0] == True
+
+        assert sum(vol[1]['a1']['volumes']) == -1
+
+    def test_good_num(self, get_visitor):
+        file = "test_cases/volume/split_good_num.bs"
+
+        tree = get_visitor(file)
+
+        vol = self.get_volume(tree, file)
+
+        assert vol[0] == False
+
+        assert sum(vol[1]['a1']['volumes']) == -1
+        assert sum(vol[1]['b1']['volumes']) == 30
+        assert vol[1]['b1']['volumes'][0] == 10
+        assert vol[1]['b1']['volumes'][1] == 10
+        assert vol[1]['b1']['volumes'][2] == 10
+
+    def test_bad_num(self, get_visitor):
+        file = "test_cases/volume/split_bad_num.bs"
+
+        tree = get_visitor(file)
+
+        vol = self.get_volume(tree, file)
+
+        assert vol[0] == True
+
+    def test_if_else(self, get_visitor):

```

Comment:
The result here does not seem correct.  There are two potential results that could come from the analysis on `dispense_if_else.bs`:

(1) if your analysis is capable of determining flow statically (as in this case 3<=3 is always true), then vol(a) = {5} prior to splitting, and null after the split, and each element of b should have b(i) = {1}.

(2) if your analysis ignores flow (minding not all conditionals can be statically determined, so this should be implemented either way when we have a variable conditional), then vol(a) = {5, 10} prior to split and null after split, and each element of b should have b(i) = {1, 2}.